### ISSUE TITLE
Heap corruption on dispatched event listener multiple remove.

### ISSUE THREAD

**Issue by sbrednikhin** [cocos2d/cocos2d-x#15540]
Calling EventDispatcher::removeEventListener multiple times with listener that is dispatched in same moment causes heap corruption.

To repeat change NewEventDispatcherTest\NewEventDispatcherTest.cpp line 199 from one line

```
_eventDispatcher->removeEventListener(listener);
```

to two lines

```
_eventDispatcher->removeEventListener(listener);
_eventDispatcher->removeEventListener(listener);
```

and run Tests->NewEventDispatcherTest->test number 14
That click on red square.

**Comment by minggo** [comment]
@sbrednikhin Did meet assert issue in Ref::release()?

**Comment by minggo** [comment]
@sbrednikhin  fixed in #15545.
